src/parser/jobs/nin/modules/Ninjutsu.tsx

```diff
@@ -28,27 +30,28 @@ export class Ninjutsu extends Module {
 	private current?: DotonCast
 	private history: DotonCast[] = []
 
-	protected override init() {
-		this.addEventHook('cast', {by: 'player', abilityId: [this.data.actions.HYOTON.id, this.data.actions.HYOTON_TCJ.id]}, () => { this.hyotonCount++ })
-		this.addEventHook('cast', {by: 'player', abilityId: this.data.actions.RABBIT_MEDIUM.id}, () => { this.rabbitCount++ })
-		this.addEventHook('cast', {by: 'player', abilityId: [this.data.actions.DOTON.id, this.data.actions.DOTON_TCJ.id]}, this.onDotonCast)
-		this.addEventHook('normaliseddamage', {by: 'player', abilityId: this.data.statuses.DOTON.id}, event => { this.current?.ticks.push(event.hitCount || 0) })
-		this.addEventHook('removebuff', {by: 'player', abilityId: this.data.statuses.DOTON.id}, this.finishDotonWindow)
+	override initialise() {
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+		this.addEventHook(playerFilter.type('action').action(oneOf([this.data.actions.HYOTON.id, this.data.actions.HYOTON_TCJ.id])), () => { this.hyotonCount++ })
+		this.addEventHook(playerFilter.type('action').action(this.data.actions.RABBIT_MEDIUM.id), () => { this.rabbitCount++ })
+		this.addEventHook(playerFilter.type('action').action(oneOf([this.data.actions.DOTON.id, this.data.actions.DOTON_TCJ.id])), this.onDotonCast)
+		this.addEventHook(playerFilter.type('damage').cause(this.data.matchCauseStatus(['DOTON'])), event => { this.current?.ticks.push(event.targets.length || 0) })
```

Comment:
```suggestion
		this.addEventHook(playerFilter.type('damage').cause(this.data.matchCauseStatus(['DOTON'])), event => { this.current?.ticks.push(event.targets.length ?? 0) })
```

We should move toward preferring ?? for nullish coalescing, rather than relying on falsey (granted, when we're coalescing a number to 0 it doesn't make any functional difference, but for making the most use of the language's type features...)